frontend/src/metabase/dashboard/components/DashboardInfoSidebar/DashboardDetails.tsx

```diff
@@ -68,7 +68,7 @@ export const DashboardDetails = ({ dashboard }: { dashboard: Dashboard }) => {
           <div>
             <Text>
               <Link
-                to={`/collection/${dashboard.collection_id}`}
+                to={`/collection/${dashboard.collection_id ?? "root"}`}
```

Comment:
I think `Urls.collection` also handles this

`frontend/src/metabase/lib/urls/collections.ts:41`